src/main/scala/org/broadinstitute/dsde/firecloud/service/NihService.scala

```diff
@@ -167,6 +170,11 @@ class NihService(val samDao: SamDAO,
         }
     }
 
+  def getNihResources(userInfo: UserInfo): Future[NihResources] =
+    getAllAllowlistGroupMemberships(userInfo).map { allowlistMembership =>
+      NihResources(allowlistMembership)
+    }
+
```

Comment:
Optional, your call: you don't actually need a new `NihResources` model object. You could have this return
```scala
NihStatus(datasetPermissions = allowlistMembership, linkedNihUsername = None, linkExpireTime = None)
```
which would result in the same API response payload to the end user.

Having a separate model object DOES make the intent clear in the code, so I am fine with either way, up to you